buildSrc/subprojects/cleanup/src/main/kotlin/gradlebuild/root-cleanup.gradle.kts

```diff
@@ -13,22 +13,26 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+package gradlebuild
 
-package gradlebuild.cleanup.tasks
-
+import gradlebuild.cleanup.extension.CleanupExtension
 import gradlebuild.cleanup.services.DaemonTracker
-import org.gradle.api.DefaultTask
-import org.gradle.api.provider.Property
-import org.gradle.api.tasks.Internal
-import org.gradle.api.tasks.TaskAction
+import gradlebuild.cleanup.tasks.KillLeakingJavaProcesses
 
+plugins {
+    base
+}
 
-abstract class CleanUpDaemons : DefaultTask() {
-    @get:Internal
-    abstract val tracker: Property<DaemonTracker>
+val trackerService = gradle.sharedServices.registerIfAbsent("daemonTracker", DaemonTracker::class) {
+    parameters.gradleHomeDir.fileValue(gradle.gradleHomeDir)
+    parameters.rootProjectDir.set(layout.projectDirectory)
+}
+extensions.create<CleanupExtension>("cleanup", trackerService)
```

Comment:
I find it a bit odd having to share the "shared" build service explicitly via an extension.
I wish there was a `Provider<BuildService> getBuildService("serviceName")` in the service registry instead.

Reply:
I agree. Is there a particular reason why we do not have that? @bamboo do you have some insights/thoughts on this maybe?